src/Symfony/Component/HttpClient/EventSourceHttpClient.php

```diff
@@ -0,0 +1,153 @@
+
+                if (null !== $chunk->getInformationalStatus()) {
+                    yield $chunk;
+
+                    return;
+                }
+            } catch (TransportExceptionInterface $e) {
+                $state->lastError = $lastError ?? microtime(true);
+
+                if (null === $state->buffer || ($isTimeout && microtime(true) - $state->lastError < $state->reconnectionTime)) {
+                    yield $chunk;
+                } else {
+                    $options['headers']['Last-Event-ID'] = $state->lastEventId;
+                    $state->buffer = '';
+                    $state->lastError = microtime(true);
+                    $context->getResponse()->cancel();
+                    $context->replaceRequest($method, $url, $options);
+                    if ($isTimeout) {
+                        yield $chunk;
+                    } else {
+                        $context->pause($state->reconnectionTime);
+                    }
+                }
+
+                return;
+            }
+
+            if ($chunk->isFirst()) {
+                if (0 === strpos($context->getHeaders()['content-type'][0] ?? '', 'text/event-stream')) {
+                    $state->buffer = '';
+                } elseif (null !== $lastError || (null !== $state->buffer && 200 === $context->getStatusCode())) {
+                    throw new EventSourceException(sprintf('Response content-type is "%s" while "text/event-stream" was expected for "%s".', $context->getHeaders()['content-type'][0] ?? '', $context->getInfo('url')));
+                } else {
+                    $context->passthru();
+                }
+
+                if (null === $lastError) {
+                    yield $chunk;
+                }
+
+                return;
+            }
+
+            $rx = "/((?:\r\n|[\r\n]){2,})/";
+            $content = $state->buffer.$chunk->getContent();
+
+            if ($chunk->isLast()) {
+                $rx = substr_replace($rx, '|$', -2, 0);
+            }
+            $events = preg_split($rx, $content, -1, PREG_SPLIT_DELIM_CAPTURE);
+            $state->buffer = array_pop($events);
+
+            for ($i = 0; isset($events[$i]); $i += 2) {
+                $event = new ServerSentEvent($events[$i].$events[1 + $i]);
+
+                if ('' !== $event->getId()) {
+                    $context->setInfo('last_event_id', $state->lastEventId = $event->getId());
+                }
+
+                if ($event->getRetry()) {
+                    $context->setInfo('reconnection_time', $state->reconnectionTime = $event->getRetry());
+                }
+
+                yield $event;
+            }
+
+            if (preg_match('/^(?::[^\r\n]*+(?:\r\n|[\r\n]))+$/m', $state->buffer)) {
+                $content = $state->buffer;
+                $state->buffer = '';
+
+                yield $context->createChunk($content);
+            }
+
+            if ($chunk->isLast()) {
+                yield $chunk;
```

Comment:
also, is it possible to reach the last chunk with some remaining buffered content in the state ? and what happens in that case ?

Reply:
I'd say that it's not really a problem and if the last chunk is partial content it won't be transformed as a `ServerSentEvent` chunk (which parses only full chunks). The full data will be retrieved later by using a `Last-Event-Id` and there's no need to process it here.

Reply:
The last chunk is always empty so that's OK. About the remaining data in the buffer, L122 above ensures it's always emptied when the last chunk arrives.